storage/directory.go

```diff
@@ -53,30 +53,30 @@ func (s *DirectoryClient) List(ctx context.Context, input *ListDirectoryInput) (
 
 	reqInput := client.RequestInput{
 		Method: http.MethodGet,
-		Path:   path,
+		Path:   fullPath,
 		Query:  query,
 	}
 	respBody, respHeader, err := s.client.ExecuteRequestStorage(ctx, reqInput)
-	if respBody != nil {
-		defer respBody.Close()
-	}
 	if err != nil {
-		return nil, errwrap.Wrapf("Error executing ListDirectory request: {{err}}", err)
+		return nil, errwrap.Wrapf("Error executing List request: {{err}}", err)
 	}
+	defer respBody.Close()
 
 	var results []*DirectoryEntry
-	for {
+	scanner := bufio.NewScanner(respBody)
+	for scanner.Scan() {
```

Comment:
Makes perfect sense since the split function scans lines by default.

Reply:
My only issue with what I did there was there's no timeout or deadline cancellation. Something like this needs to be integrated still: https://gist.github.com/dchapes/6c992bf3e943934462509338cd213e99